include/led.h

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2018 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _ZEPHYR_LED_H
+#define _ZEPHYR_LED_H
+
+/**
+ * @file
+ * @brief Public LED driver APIs
+ */
+
+#include <zephyr/types.h>
+#include <device.h>
+
+/**
+ * @brief Internal driver specific representation of an LED device
+ */
+
+struct led_data {
+	/* Minimum acceptable LED blinking time period (in ms) */
+	u32_t min_period;
+	/* Maximum acceptable LED blinking time period (in ms) */
+	u32_t max_period;
+	/* Minimum acceptable LED brightness value */
+	u8_t min_brightness;
```

Comment:
nit: I worked on a project in the past where we needed a 10,000:1 PWM duty cycle range for a button LED at it needed to be visible in daylight and dim enough at night to not affect night vision.

Consider changing this to u16_t?

Reply:
Sure. u16_t sounds reasonable.